### PR TITLE
Add new leaves style - simple (glasslike drawtype)

### PR DESCRIPTION
**Pull request by RealBadAngel**

### CODE REVIEW

File: minetest.conf.example [hunk 1]
```diff
@@ -108,8 +108,9 @@
 #liquid_queue_purge_time = 0
 #    Liquid update interval in seconds
 #liquid_update = 1.0
-#    Enable transparent leaf textures, disable for speed
-#new_style_leaves = true
+#    Leaves style: fancy - all faces visible, simple - only outer faces,
+#    opaque - disable transparency
```
Comment (asl97) [inline minetest.conf.example]: my first look at this is a little confusing, maybe indent it or split it up like https://github.com/RealBadAngel/minetest/blob/d7a2523df82b2452726b4ef51d3b8ff7ca558636/minetest.conf.example#L159

Comment (est31) [inline minetest.conf.example]: makes sense

File: builtin/mainmenu/tab_settings.lua [hunk 2]
```diff
@@ -39,6 +50,16 @@ local mipmap = {
 	{"", "mip_map", "anisotropic_filter"},
 }
 
+local function getLeavesStyleSettingIndex()
+	if (core.setting_get("leaves_style") == leaves_style[2][3]) then
```
Comment (est31) [inline builtin/mainmenu/tab_settings.lua]: You shouldn't call core.setting_get twice, but rather cache the value.

File: builtin/mainmenu/tab_settings.lua [hunk 3]
```diff
@@ -353,7 +374,18 @@ local function handle_settings_buttons(this, fields, tabname, tabdata)
 		core.setting_set("touchscreen_threshold",fields["dd_touchthreshold"])
 		ddhandled = true
 	end
-
+	if fields["dd_leaves_style"] == leaves_style_labels[3] then
+		core.setting_set("leaves_style", leaves_style[2][3])
+		ddhandled = true
+	end
+	if fields["dd_leaves_style"] == leaves_style_labels[2] then
```
Comment (est31) [inline builtin/mainmenu/tab_settings.lua]: These should be elseif cases.

**Comment by est31** [comment]
Glasslike:
screenshot_glasslike
Opaque:
screenshot_opaque
Fancy:
screenshot_fancy
You see that there are no noticeable FPS differences for me.
In my opinion, fancy looks best (what suprise), followed by opaque. Glasslike isn't very believeable IMO, because you don't have any "denseness". The trees look like sticks with very light leaves around. More complex trees (like those in dreambuilder) don't look worse though.

**Comment by RealBadAngel** [comment]
Just for the record: simple leaves with an example altered texture:
screenshot_20150708_031257

**Comment by RealBadAngel** [comment]
https://github.com/minetest/minetest/commit/39439cbd3dd4acce0ff897e3e277736251dde2a4

**Comment by paramat** [comment]
Glasslike looks okay with that really dense texture.

**Pull request closed without merging**